src/DynamicData.Tests/Binding/AvaloniaDictionaryFixture.cs

```diff
@@ -0,0 +1,303 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.ComponentModel;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+
+using  DynamicData.Binding;
+using DynamicData.Tests.Domain;
+using FluentAssertions;
+using Xunit;
+
+namespace DynamicData.Tests.Binding;
+
+public class AvaloniaDictionaryFixture
+{
+    private readonly AvaloniaDictionary<string, Person> _collection;
+    private readonly ChangeSetAggregator<Person> _results;
+
+    public AvaloniaDictionaryFixture()
+    {
+        _collection = new AvaloniaDictionary<string, Person>();
+        _results =  _collection.ToObservableChangeSet<AvaloniaDictionary<string, Person>, KeyValuePair<string, Person>>()
+            .Transform(x=>x.Value)
+            .AsAggregator();
+    }
+
+    [Fact]
+    public void Add()
+    {
+        var person = new Person("Someone",10, "M");
+
+        _collection.Add("Someone", person);
+
+        _results.Messages.Count.Should().Be(1);
+        _results.Data.Count.Should().Be(1);
+        _results.Data.Items.First().Should().Be(person);
+    }
+
+    [Fact]
+    public void Replace()
+    {
+        var person1 = new Person("Someone", 10, "M");
+        var person2 = new Person("Someone", 11, "M");
+
+        _collection.Add("Someone", person1);
+        _collection["Someone"] = person2;
+
+        _results.Data.Count.Should().Be(1);
+        _results.Data.Items.First().Should().Be(person2);
+    }
+
+
+    [Fact]
+    public void Remove()
+    {
+        var person = new Person("Someone", 10, "M");
+
+        _collection.Add("Someone", person);
+        _collection.Remove(person.Key);
+
+        _results.Data.Count.Should().Be(0);
+    }
+}
+
+
+public interface IAvaloniaDictionary<TKey, TValue>
+    : IDictionary<TKey, TValue>,
+        IAvaloniaReadOnlyDictionary<TKey, TValue>,
+        IDictionary
+    where TKey : notnull
+{
+}
+
+
+public interface IAvaloniaReadOnlyDictionary<TKey, TValue>
+    : IReadOnlyDictionary<TKey, TValue>,
+        INotifyCollectionChanged,
+        INotifyPropertyChanged
+    where TKey : notnull
+{
+}
+
+public class AvaloniaDictionary<TKey, TValue> : IAvaloniaDictionary<TKey, TValue> where TKey : notnull
```

Comment:
I copied this from the AvaloniaProject so we can test it,

Reply:
A little weird, but okay. Me,

I think what I would have done is just improve coverage, in general, on the fixture for this `ToObservableChangeSet()` operator. To me, the core issue is that the operator wasn't compatible with the full functionality of `INotifyCollectionChanged`, `AvaloniaDictionary` just happens to be one of the implementations that leverages functionality we don't support. I.E. 

I will say you should add a comment here, mentioning that this is copied directly form Avalonia, and why.

Reply:
With the standard observable collection these scenarios cannot be reproduced - period.  I had covered everything on earth that MSFT produced but specifically the Avalonia Dictionary give alternative results.

The only other viable reproduction would be to re-create the Avalonia dictionary in all but name, so I may as well use the real thing.